### PR TITLE
#174 - localize vue

### LINKED ISSUE #128: Search Form in the Company Browser
As in title, right now the component shows all the companies provided in file.

### LINKED ISSUE #174: Localize strings in vue templates
It would be nice if the website received multilanguage support. It'd make it more accessible.

Initially, we are going to implement two languages:
- Polish
- English

Localization implementation we'll decide on should make it easier to add new languages.

### PR DESCRIPTION
**Pull request by MatArasim**
This PR implements multilanguage support, for now Polish and English. The design of the 'switch language button' is going to be changed later after main website functions get implemented.

image

image

The responsive filters tab was added to the search, it expands:

image

image

Also introduced a new side slider that shows information about selected company:

image

Closes #174, closes #128.

### CODE REVIEW

File: frontend/components/NavigationBar.vue [hunk 1]
```diff
@@ -15,12 +15,12 @@
...
                 class="bg-gray-900 text-white px-3 py-2 rounded-md text-sm font-medium"
-                >Mapa</a
-              >
+                >{{ $t("NavigationBar.map") }}
+              </a>
               <a
                 href="#"
                 class="text-gray-300 hover:bg-gray-700 hover:text-white px-3 py-2 rounded-md text-sm font-medium"
-                >Dodaj firmę</a
+                >{{ $t("NavigationBar.add-company") }}</a
```
Comment (EwelinaLasowy) [inline frontend/components/NavigationBar.vue:23]: Can we unify this to form example `snake_case`? For now we have `PascalCase` and `kebab-case` and it's not comfortable to read this.

File: frontend/components/FilterDisclosure.vue [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+          <ChevronDownIcon
+            :class="open ? 'rotate-180 transform' : ''"
+            class="h-5 w-5 text-slate-500"
+          />
+        </DisclosureButton>
+        <DisclosurePanel class="px-4 pt-4 pb-2 text-sm text-slate-500">
+          <FilterCombobox
+            :placeholder="$t('Comboboxes.combobox-city')"
+            :items="cities"
+            :id="'cities-${index}'"
```
Comment (EwelinaLasowy) [inline frontend/components/FilterDisclosure.vue:18]: Do we really need this `index`? IN this component there is no loop.

File: frontend/components/FilterDisclosure.vue [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+<script setup>
+import { Disclosure, DisclosureButton, DisclosurePanel } from "@headlessui/vue";
+import { ChevronDownIcon } from "@heroicons/vue/solid";
+import FilterCombobox from "@/components/ComboBox.vue";
+
+const cities = ["City", "Other city"];
+const specialization = ["A specialization", "Other specialization"];
+const studyfields = ["Some studyfields", "Other studyfields"];
+const depreparments = ["Some depertment", "Other department"];
```
Comment (EwelinaLasowy) [inline frontend/components/FilterDisclosure.vue:51]: A typo. Remember to fix this everywhere.
```suggestion
const departments = ["Some department", "Other department"];
```

File: frontend/components/CompanyList.vue [hunk 4]
```diff
@@ -2,44 +2,118 @@
...
+  {
+    id: "2",
+    name: "Some Company",
+    email: "some.company@example.com",
+    image: PlaceholderImage,
+    description:
+      "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nullam consequat nisi vitae tortor accumsan, ac maximus leo elementum. Praesent vestibulum dapibus eros at vestibulum. Fusce ut neque velit. Proin pharetra dui sit amet magna eleifend mattis. Nunc eget tortor vitae arcu condimentum rutrum. Sed molestie est justo, nec placerat purus pharetra nec. Morbi interdum elementum dignissim. Mauris eleifend pellentesque ligula quis faucibus. Suspendisse sit amet sodales risus, sit amet gravida est. Vivamus ac pharetra eros. Mauris nisi justo, semper quis libero id, tincidunt maximus nisi. Nullam cursus tempus orci eget rhoncus. Praesent elementum pretium purus, eget dictum magna porttitor quis. Suspendisse ac commodo risus. Praesent non ullamcorper massa, sit amet pellentesque augue.",
+  },
+  {
+    id: "1",
```
Comment (EwelinaLasowy) [inline frontend/components/CompanyList.vue:39]: I know, that is some kind of data stub, but please make id unique for consistency.

File: frontend/components/LanguageSwitch.vue [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+        :value="locale.lang"
+      >
+        {{ locale.name }}
+      </option>
+    </select>
+  </div>
+</template>
+
+<script setup>
+const locales = [
```
Comment (EwelinaLasowy) [inline frontend/components/LanguageSwitch.vue:20]: Can't we move this to translations?

File: frontend/components/CompanyList.vue (hunk 4 above)
[the remark this answers is not in the document]
Comment (MatArasim) [reply frontend/components/CompanyList.vue]: Done, now all placeholders companies have their own unique id

File: frontend/components/FilterDisclosure.vue (hunk 2 above)
Comment (MatArasim) [reply frontend/components/FilterDisclosure.vue:18]: It wasnt need it, already removed

File: frontend/components/LanguageSwitch.vue (hunk 5 above)
Comment (MatArasim) [reply frontend/components/LanguageSwitch.vue]: Translations was moved to `index.js` in `assets/lang`

**Review by EwelinaLasowy — APPROVED** [review]

**Review by dawidrudnik — APPROVED** [review]

**Pull request merged**